modules/social_features/social_album/social_album.module

```diff
@@ -0,0 +1,351 @@
+      $items[$pid] = ['post' => $parent_variables['post']];
+    }
+
+    $items[$pid]['urls'][] = $item['url'];
+  }
+
+  $variables['items'] = $items;
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_insert().
+ */
+function social_album_post_insert(EntityInterface $entity) {
+  /** @var \Drupal\social_post\Entity\PostInterface $entity */
+  if ($entity->bundle() === 'photo' && !$entity->field_album->isEmpty()) {
+    \Drupal::service('cache_tags.invalidator')->invalidateTags([
+      'node:' . $entity->field_album->entity->id(),
+      'user:' . $entity->getOwnerId(),
+    ]);
+  }
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_create_access().
+ */
+function social_album_node_create_access(AccountInterface $account, array $context, $entity_bundle) {
+  if (
+    $entity_bundle === 'album' &&
+    \Drupal::routeMatch()->getRouteName() === 'view.albums.page_albums_overview' &&
+    \Drupal::routeMatch()->getRawParameter('user') !== $account->id()
+  ) {
+    return AccessResult::forbidden();
+  }
+
+  return AccessResult::neutral();
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_insert().
+ */
+function social_album_node_insert(EntityInterface $entity) {
+  /** @var \Drupal\node\NodeInterface $entity */
+  if ($entity->bundle() === 'album') {
+    \Drupal::service('cache_tags.invalidator')->invalidateTags([
+      'user:' . $entity->getOwnerId(),
+    ]);
+  }
+}
+
+/**
+ * Implements hook_social_post_message_alter().
+ */
+function social_album_social_post_message_alter(TranslatableMarkup &$message, FormStateInterface $form_state) {
+  if ($form_state->has('album')) {
+    $post = $form_state->getFormObject()->getEntity();
+
+    if (!$post->field_album->isEmpty()) {
+      $message = t('You have posted successfully in your new album @link', [
+        '@link' => $post->field_album->entity->toLink()->toString(),
+      ]);
+    }
+  }
+}
+
+/**
+ * Implements hook_entity_type_alter().
+ */
+function social_album_entity_type_alter(array &$entity_types) {
+  /** @var \Drupal\Core\Entity\EntityTypeInterface[] $entity_types */
+  $entity_types['post']
+    ->setFormClass('album', SocialAlbumPostForm::class)
+    ->setFormClass('delete_image', SocialAlbumImageForm::class);
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function social_album_form_comment_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $route_name = \Drupal::routeMatch()->getRouteName();
+
+  if ($is_image_view = $route_name === 'social_album.image.view') {
+    /** @var \Drupal\comment\CommentInterface $comment */
+    $comment = $form_state->getFormObject()->getEntity();
+
+    $form['actions']['submit']['#attributes']['class'][] = 'post-index-' . $comment->getCommentedEntityId();
+
+    if ($first = &drupal_static(__FUNCTION__, TRUE)) {
+      $first = FALSE;
+    }
+    else {
+      $form['actions']['submit']['#attributes']['class'][] = 'hide';
+    }
+  }
+
+  if (!(
+    !\Drupal::moduleHandler()->moduleExists('social_ajax_comments') ||
```

Comment:
In this case, I think we can use variables (`$route_match` and `$module_handler` because it calls twice.